src/Orleans.Core/Messaging/CachingIdSpanCodec.cs

```diff
@@ -87,21 +89,11 @@ public IdSpan ReadRaw<TInput>(ref Reader<TInput> reader)
         private void PurgeStaleEntries()
         {
             var currentTimestamp = Environment.TickCount64;
-            List<int> purgeKeys = default;
             foreach (var entry in _cache)
             {
                 if (currentTimestamp - entry.Value.LastSeen > PurgeAfterMilliseconds)
                 {
-                    purgeKeys ??= new();
-                    purgeKeys.Add(entry.Key);
-                }
-            }
-
-            if (purgeKeys is not null)
-            {
-                foreach (var key in purgeKeys)
-                {
-                    _cache.Remove(key);
+                    _cache.Remove(entry.Key);
```

Comment:
There are likely other places where we're using this pattern which is now unnecessary